packages/xo-server/src/xapi/mixins/vm.js

```diff
@@ -82,7 +84,18 @@ export default {
     )
     $defer.onFailure(() => this.deleteVm(vmRef))
 
-    // TODO: copy BIOS strings?
+    let vm = await this._getOrWaitObject(vmRef)
+
+    const isHvm = isVmHvm(vm)
+
+    // Copy BIOS strings
+    if (isHvm && copyHostBiosStrings && props.affinityHost !== undefined) {
+      await this.callAsync(
+        'VM.copy_bios_strings',
```

Comment:
What if the template has already copied `BIOS strings` from host.

Reply:
it will generate an error.
 the `BIOS strings` can be set once.

Reply:
Should the `VM` creation be interrupted if the template has already bios strings?

Reply:
IMHO, yes. If the user doesn' t need it, the `copyHostBiosStrings` will be `false`.
What do you think?

Reply:
But if the user need it and the template had already copied the bios strings the VM creation will fails.
IMHO, we could inform the user using the UI that the template had already copied the bios strings.